app/questionnaire/navigator.py

```diff
@@ -77,29 +76,24 @@ def build_path(self, blocks, this_location):
             # If routing rules exist then a rule must match (i.e. default goto)
             if 'routing_rules' in block and len(block['routing_rules']) > 0:
                 original_this_location = this_location.copy()
-                for rule in block['routing_rules']:
-                    if SchemaHelper.is_goto_rule(rule) and evaluate_goto(rule['goto'], self.metadata, self.answer_store, this_location['group_instance']):
-                        is_meta_rule = SchemaHelper.is_goto_meta_rule(rule)
+                for rule in [r for r in block['routing_rules'] if SchemaHelper.is_goto_rule(r)]:
```

Comment:
If these `@staticmethod`s in SchemaHelper were just top-level methods on the schema_helper.py this could be even easier to read:

```Python
for rule in filter(is_goto_rule, block['routing_rules']):
```